src/lib/swapper/swappers/ThorchainSwapper/utils/addAggregatorAndDestinationToMemo.test.ts

```diff
@@ -49,82 +37,83 @@ describe('addAggregatorAndDestinationToMemo', () => {
+    const memo = `=:ETH.ETH:${REALLY_BIG_ADDRESS}:${expectedL1AmountOut}:ss:${affiliateBps}`
+
+    expect(() =>
+      addAggregatorAndDestinationToMemo({
+        sellChainId: btcChainId,
+        quotedMemo: memo,
+        aggregator: AGGREGATOR_ADDRESS,
+        finalAssetAddress: FINAL_ASSET_ADDRESS,
+        minAmountOut,
+        slippageBps,
+        longtailTokens: DEFAULT_THORCHAIN_ASSETS_LIST,
+      }),
+    ).toThrow()
+  })
 
-    const slippageBps = 100 // 1%
+  it('should throw if chainId is DOGE and bytes length is > 80', () => {
+    const minAmountOut = '2083854765519275828179229'
+    const memoOver80Bytes = `=:ETH.ETH:${REALLY_BIG_ADDRESS}:${expectedL1AmountOut}:ss:${affiliateBps}`
 
     expect(() =>
       addAggregatorAndDestinationToMemo({
         sellChainId: dogeChainId,
-        quotedMemo: memoOver220Bytes,
+        quotedMemo: memoOver80Bytes,
         aggregator: AGGREGATOR_ADDRESS,
         finalAssetAddress: FINAL_ASSET_ADDRESS,
         minAmountOut,
         slippageBps,
-        finalAssetPrecision: 18,
-        chainNamespace: CHAIN_NAMESPACE.Utxo,
+        longtailTokens: DEFAULT_THORCHAIN_ASSETS_LIST,
       }),
     ).toThrow()
   })
 
-  it('should not throw if chainNamespace is UTXO and chainId is BCH', () => {
+  it('should be successful if chainId is DOGE and bytes length is > 80', () => {
     const minAmountOut = '2083854765519275828179229'
-    const affiliateBps = '100'
-    const expectedL1AmountOut = '42' // we don't care about this for the purpose of tests
-    const quotedMemo = `=:ETH.ETH:${RECEIVE_ADDRESS}:${expectedL1AmountOut}:ss:${affiliateBps}:${AGGREGATOR_TWO_LAST_CHARS}:${FINAL_ASSET_ADDRESS}:10`
-
-    const slippageBps = 100 // 1%
+    const quotedMemo = `=:ETH.ETH:${RECEIVE_ADDRESS}:${expectedL1AmountOut}:ss:${affiliateBps}`
 
     const modifiedMemo = addAggregatorAndDestinationToMemo({
-      sellChainId: bchChainId,
+      sellChainId: dogeChainId,
       quotedMemo,
       aggregator: AGGREGATOR_ADDRESS,
       finalAssetAddress: FINAL_ASSET_ADDRESS,
       minAmountOut,
       slippageBps,
-      finalAssetPrecision: 18,
-      chainNamespace: CHAIN_NAMESPACE.Utxo,
+      longtailTokens: DEFAULT_THORCHAIN_ASSETS_LIST,
     })
 
+    expect(modifiedMemo.length).toBeLessThanOrEqual(UTXO_MAXIMUM_BYTES_LENGTH)
+
     expect(modifiedMemo).toBe(
-      `=${MEMO_PART_DELIMITER}e${MEMO_PART_DELIMITER}${RECEIVE_ADDRESS}${MEMO_PART_DELIMITER}${expectedL1AmountOut}${MEMO_PART_DELIMITER}ss${MEMO_PART_DELIMITER}${affiliateBps}${MEMO_PART_DELIMITER}${AGGREGATOR_TWO_LAST_CHARS}${MEMO_PART_DELIMITER}${FINAL_ASSET_ADDRESS}${MEMO_PART_DELIMITER}206301621786412`,
+      `=${MEMO_PART_DELIMITER}e${MEMO_PART_DELIMITER}${RECEIVE_ADDRESS}${MEMO_PART_DELIMITER}${expectedL1AmountOut}${MEMO_PART_DELIMITER}ss${MEMO_PART_DELIMITER}${affiliateBps}${MEMO_PART_DELIMITER}${AGGREGATOR_TWO_LAST_CHARS}${MEMO_PART_DELIMITER}${SHORTENED_FINAL_ASSET_ADDRESS}${MEMO_PART_DELIMITER}20630162116`,
```

Comment:
Perhaps add a comment in all these explaining what is this limit in thor base unit? (i.e precision 8) since the magic exponent thingie is not very obvious to visually parse?

Reply:
d16cfe193fe